crates/interpreter/src/instructions/bitwise.rs

```diff
@@ -102,33 +107,32 @@ pub fn sar<H: Host + ?Sized, SPEC: Spec>(interpreter: &mut Interpreter, _host: &
     gas!(interpreter, gas::VERYLOW);
     pop_top!(interpreter, op1, op2);
 
-    let value_sign = i256_sign_compl(op2);
-
-    // If the shift count is 255+, we can short-circuit. This is because shifting by 255 bits is the
-    // maximum shift that still leaves 1 bit in the original 256-bit number. Shifting by 256 bits or
-    // more would mean that no original bits remain. The result depends on what the highest bit of
-    // the value is.
-    *op2 = if value_sign == Sign::Zero || op1 >= U256::from(255) {
-        match value_sign {
-            // value is 0 or >=1, pushing 0
-            Sign::Plus | Sign::Zero => U256::ZERO,
-            // value is <0, pushing -1
-            Sign::Minus => U256::MAX,
-        }
+    let shift = usize::try_from(op1).unwrap_or(256); // If op1 is greater than the maximum for usize, consider the shift complete.
```

Comment:
Please use `as_usize_saturated`, U256::try_into does not have great codegen.

Reply:
Cool, let me do it